packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -170,22 +169,32 @@ class CachedArtifacts extends Artifacts {
       case TargetPlatform.ios:
         return _getIosArtifactPath(artifact, platform, mode);
       case TargetPlatform.darwin_x64:
+        return _getDarwinArtifactPath(artifact, platform, mode);
       case TargetPlatform.linux_x64:
-      case TargetPlatform.windows_x64:
       case TargetPlatform.fuchsia:
+      case TargetPlatform.windows_x64:
       case TargetPlatform.tester:
       case TargetPlatform.web_javascript:
-        return _getHostArtifactPath(artifact, platform, mode);
+      default: // could be null, but that can't be specified as a case.
+        return _getHostArtifactPath(artifact, platform ?? _currentHostPlatform, mode);
     }
-    assert(false, 'Invalid platform $platform.');
-    return null;
   }
 
   @override
   String getEngineType(TargetPlatform platform, [ BuildMode mode ]) {
     return fs.path.basename(_getEngineArtifactsPath(platform, mode));
   }
 
+  String _getDarwinArtifactPath(Artifact artifact, TargetPlatform platform, BuildMode mode) {
+    // When platform is null, we're asking for a generic host platform artifact
```

Comment:
s/we're asking for a [...]/[...] is being requested/

Reply:
Fixed